### DISCUSSION TITLE
minio: Error: Bad sha256: Expected

### DISCUSSION THREAD

**Discussion by waem14** [minio/minio#13234] [Q&A]
## Current Behavior

Sometimes, we found this error in log.
Sep 17 17:21:02 server minio: Error: Bad sha256: Expected 1170bead4815171e0fdd67d9a0c28c4c96da900a10811392eaa6cc74e559a8bf does not match calculated 5f1af8206479e03487d177b35f621487c9ee10058a29f21709ea0ea254bab05f (hash.SHA256Mismatch)

## Steps to Reproduce (for bugs)

Don't know the root cause and how to re-produce

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration:
* Operating System and version (`uname -a`):

minio version RELEASE.2021-09-03T03-56-13Z

Linux sg-bimin-6z01 3.10.0-1127.19.1.el7.x86_64 #1 SMP Tue Aug 25 17:23:54 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by klauspost** [comment]
@waem14 It means the SHA256 hash sent by the client on upload does not match the hash of the received content and the upload was rejected.

**Reply by waem14** [reply]
Could you please explain more detail and give me some example about this scenario?

**Reply by klauspost** [reply]
When you upload a file, you can set a hash in `x-amz-content-sha256` which should reflect the content being uploaded.

If it doesn't match the data, the content is rejected.
